### PR TITLE
joint_trajectory_controller: making it so the URDF parameter is configurable

### PR DESCRIPTION
**Pull request by jbohren**
This is necessary when using this controller in simulation with multiple robots.

### CODE REVIEW

**Comment by adolfo-rt** [comment]
How's your namespace layout when using multiple robots?. I believe that the following layout should work with the current implementation:

```
/
|
|-- one_robot
|   |
|   |-- robot_description
|   |-- controller_manager
|
|-- other_robot
|   |
|   |-- robot_description
|   |-- controller_manager
```

For each robot, the `root_nh` would be `one_robot` and `other_robot`, respectively, so the URDF should be found.

I'm interested in this usecase, because I'm presently coding an rqt plugin for this controller, and I need both controller_manager and URDF info. It would be a big time saver if a sane convention could be assumed.

File: joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller_impl.h [hunk 1]
```diff
@@ -113,7 +113,20 @@ std::vector<UrdfJointConstPtr> getUrdfJoints(const urdf::Model& urdf, const std:
...
+          link_it != links.end();
+          ++link_it)
+      {
+        if((*link_it)->parent_joint) {
+          joints_oss << " - \"" << (*link_it)->parent_joint->name << "\"\n";
+        }
+      }
+      ROS_ERROR_STREAM(
+          "Could not find joint '" << joint_names[i] << "' in URDF model.\n"
+          << "Available joints include: \n" << joints_oss.str());
```
Comment (adolfo-rt) [inline joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller_impl.h:129]: This is an extra treat! ;-)

**Comment by adolfo-rt** [comment]
Bump.
@jbohren, could you please comment on my [past comment](https://github.com/ros-controls/ros_controllers/pull/98#issuecomment-44458604)?. I'm interested in understanding your usecase.

**Comment by adolfo-rt** [comment]
Bump again.

**Comment by jbohren** [comment]
So this was specifically a problem with simulation, it should be resolved by using the `<robotNamespace>` tag in the `gazebo_ros_control` plugin, but for some reason that didn't work. If that should work, then maybe that's where the bug is.

**Comment by adolfo-rt** [comment]
Long time no change.

**Pull request closed without merging**